ng2-components/ng2-activiti-form/src/components/widgets/attach/attach.widget.spec.ts

```diff
@@ -0,0 +1,293 @@
+                    selectedFolder: config
+                }
+            }
+        });
+        widget.ngOnInit();
+
+        expect(widget.selectedFolderSiteId).toBe(config.siteId);
+        expect(widget.selectedFolderSiteName).toBe(config.site);
+        expect(widget.selectedFolderPathId).toBe(config.pathId);
+        expect(widget.selectedFolderAccountId).toBe(config.accountId);
+        expect(widget.selectedFolderNodes).toEqual(nodes);
+    });
+
+    it('should link file on select', () => {
+        let link = <ExternalContentLink> {};
+        spyOn(contentService, 'linkAlfrescoNode').and.callFake(() => {
+            return Observable.create(observer => {
+                observer.next(link);
+                observer.complete();
+            });
+        });
+
+        widget.field = new FormFieldModel(null, {
+            type: FormFieldTypes.UPLOAD
+        });
+        widget.ngOnInit();
+
+        let node = <ExternalContent> {};
+        widget.selectFile(node, null);
+
+        expect(contentService.linkAlfrescoNode).toHaveBeenCalled();
+        expect(widget.selectedFile).toBe(node);
+        expect(widget.field.value).toEqual([link]);
+        expect(widget.field.json.value).toEqual([link]);
+    });
+
+    it('should reset', () => {
+        widget.hasFile = true;
+        widget.field = new FormFieldModel(null, {
+            type: FormFieldTypes.UPLOAD,
+            value: [{ name: 'filename' }]
+        });
+
+        widget.reset();
+        expect(widget.hasFile).toBeFalsy();
+        expect(widget.field.value).toBeNull();
+        expect(widget.field.json.value).toBeNull();
+    });
+
+    it('should close dialog on cancel', () => {
+        let closed = false;
+        widget.dialog = {
+            nativeElement: {
+                close: function() {
+                    closed = true;
+                }
+            }
+        };
+        widget.cancel();
+        expect(closed).toBeTruthy();
+    });
+
+    it('should show modal dialog', () => {
+        spyOn(contentService, 'getAlfrescoNodes').and.callFake(() => {
+            return Observable.create(observer => {
+                 observer.next([]);
+                 observer.complete();
+            });
+        });
+
+        widget.field = new FormFieldModel(null, {
+            type: FormFieldTypes.UPLOAD,
+            params: {
+                fileSource: {
+                    selectedFolder: {}
+                }
+            }
+        });
+
+        let modalShown = false;
+        widget.dialog = {
+            nativeElement: {
+                showModal: function() {
+                    modalShown = true;
+                }
+            }
+        };
+
+        widget.showDialog();
+        expect(modalShown).toBeTruthy();
+    });
+
+    it('should select folder and load nodes', () => {
+        let nodes = [{}];
+        spyOn(contentService, 'getAlfrescoNodes').and.callFake(() => {
+            return Observable.create(observer => {
```

Comment:
Could `Observer.from()` simplify the creation? And possibly spy's `returnValue` rather than `callFake()`?